File: Algorithms/Greedy/PerfectChange.java

public class PerfectChange {

    public static int getNumCoins(int[] values, int amt) {

        int coins = 0;

        for (int v : values) {

            while (amt >= v) {

                amt -= v;
                coins++;

            }

        }

        return coins;

    }

    public static void main(String[] args) {

        int[] values = new int[]{10, 5, 1};

        System.out.println(getNumCoins(values, 28));

    }

}

File: Algorithms/Greedy/PerfectChange.js
"use strict";

let getNumCoins = (values, amt) => {

    let coins = 0;

    for (let v of values) {

        while (amt >= v) {

            amt -= v;
            coins++;

        }

    }

    return coins;

}

let values = [10, 5, 1];

console.log(getNumCoins(values, 28));
